### ISSUE TITLE
ImportError: No module named scapy.all

### ISSUE THREAD

**Issue by codebyAbhishekBharti** [DanMcInerney/wifijammer#111]
When i am trying to execute the file using the the command:
**sudo python wifijammer** also tried using **sudo python2 wifijammer**
But it is returning the error:
**Traceback (most recent call last):
  File "wifijammer", line 6, in <module>
    from scapy.all import *
ImportError: No module named scapy.all**

but when I run the  **from scapy.all import** *   in *python interpreter* it is working fine.

*SYSTEM DETAILS*
OS: Linux mint 20
Python 2.7.18

**Comment by acidOP** [comment]
pip install --pre scapy[basic]
This might help. Do let me know if it works.

**Comment by codebyAbhishekBharti** [comment]
> pip install --pre scapy[basic]
> This might help. Do let me know if it works.

@acidOP  Not working at all dear. Still getting `ImportError: No module named scapy.all
`
Fun fact:-
If i am simply creating a python file `check.py` with just `from scapy.all import *` code in it and then run the file with `python2 check.py` then the module is getting imported successfully but if i apply `sudo python2 check.py` then it is saying the error `ImportError: No module named scapy.all
`
In short error is happening only when i am applying **sudo** to open the file else module is getting imported. But wifi jammer don't run without **sudo**.

**Comment by acidOP** [comment]
Try installing scapy with sudo.

**Comment by codebyAbhishekBharti** [comment]
> Try installing scapy with sudo.

If i am trying to install **scapy** with **sudo** using the command `sudo pip install scapy` it is saying `sudo: pip: command not found
` where as if i am trying `pip install scapy` then it is saying `Requirement already satisfied: scapy in /home/abhishek/.local/lib/python2.7/site-packages (2.4.4)`

**Comment by jubernar1993** [comment]
Did you find a fix for the issue? I’m stuck with that same error!

**Comment by oldkingcone** [comment]
python3 -m pip install scapy --user

python2 -m pip install scapy --user

$( which pip3 ) install scapy --user

$( which pip2 ) install scapy --user

$( which easy_install ) scapy

**Comment by oldkingcone** [comment]
what does `$( which pip3 ) list | grep scapy` produce

**Comment by irtizarana** [comment]
I have also same error , plz help us

**Comment by AhmadVakil** [comment]
Same problem here...

**Comment by JustVladKaz** [comment]
Solution was in 16th issue!

thx @gconelhero - I just copy & paste...
Here it is: 

Digite no terminal:
`sudo mkdir /usr/lib/python2.7/dist-packages/scapy`
`cd /usr/lib/python3/dist-packages/`
`cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`

Resolvido!

Type in the terminal:
`sudo mkdir /usr/lib/python2.7/dist-packages/scapy`
`cd /usr/lib/python3/dist-packages/`
`cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`

Solved!

**Comment by Fabriyin** [comment]
> Solution was in 16th issue!
> 
> thx @gconelhero - I just copy & paste... Here it is:
> 
> Digite no terminal: `sudo mkdir /usr/lib/python2.7/dist-packages/scapy` `cd /usr/lib/python3/dist-packages/` `cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`
> 
> Resolvido!
> 
> Type in the terminal: `sudo mkdir /usr/lib/python2.7/dist-packages/scapy` `cd /usr/lib/python3/dist-packages/` `cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`
> 
> Solved!


I tried this but didn't help, now I tried again but using "sudo" and finally worked

**Comment by asquzerr** [comment]
> Solution was in the 16th issue!
> 
> thx @gconelhero - I just copy & paste... Here it is:
> 
> Digite no terminal: `sudo mkdir /usr/lib/python2.7/dist-packages/scapy` `cd /usr/lib/python3/dist-packages/` `cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`
> 
> Resolvido!
> 
> Type in the terminal: `sudo mkdir /usr/lib/python2.7/dist-packages/scapy` `cd /usr/lib/python3/dist-packages/` `cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`
> 
> Solved!

I tried the same thing but did not work for me, do you have another solution?
every time I try to execute the python file with the scapy package he gives me this output:
  File "/home/kali/Desktop/python/scapy.py", line 2, in <module>
    from scapy.all import *
  File "/home/kali/Desktop/python/scapy.py", line 2, in <module>
    from scapy.all import *
ModuleNotFoundError: No module named 'scapy.all'; 'scapy' is not a package

**Comment by elrafinha64** [comment]
> > pip install --pre scapy[basic]
> > This might help. Do let me know if it works.
> 
> @AcidOP Not working at all dear. Still getting `ImportError: No module named scapy.all ` Fun fact:- If i am simply creating a python file `check.py` with just `from scapy.all import *` code in it and then run the file with `python2 check.py` then the module is getting imported successfully but if i apply `sudo python2 check.py` then it is saying the error `ImportError: No module named scapy.all ` In short error is happening only when i am applying **sudo** to open the file else module is getting imported. But wifi jammer don't run without **sudo**.

I have exactly this same issue, Did you solved it?

**Comment by asquzerr** [comment]
Hi, how are you?
I didn't solved it yet there are problem in python2 with scapy model i
didnot understand the problem

**Comment by gabeh33** [comment]
I had this problem because I was naming my file scapy.py and trying to run it using python scapy.py. That didn't work, but changing the name to just scapy, making it executable, and just running it with ./scapy seems to work

**Comment by AsquzeRR** [comment]
Heyy bro, i solved it by defining the scapy manually i find this soultion
with my experience it was on windows not on kali linux, the problem is the
python cannot define the sxapy.all as a package and it was installed, in as
i read in that while happend a problem in scapy in windows i found a
 misconfiguration. Thanks bro❤️

**Comment by AcidOP** [comment]
@AsquzeRR Hey you mind creating a PR solving the misconfiguration? 😀

**Comment by Dwebb22** [comment]
Tried nothing worked. Hey you have anything for start root?

Dee

**Comment by gconelhero** [comment]
> Same problem here...
`sudo mkdir /usr/lib/python2.7/dist-packages/scapy
cd /usr/lib/python3/dist-packages/
cp -avr scapy/* /usr/lib/python2.7/dist-packages/scapy`

**Comment by gconelhero** [comment]
> 

What operating system are you working on?

**Comment by AsquzeRR** [comment]
https://github.com/DanMcInerney/wifijammer/issues/111#issuecomment-1446313682
Solved!

**Comment by anselmobattisti** [comment]
I`m in a ubuntu 20.02 and running the script using virtual environment venv

after running inside the venv

`pip install scapy `

i got "ModuleNotFoundError: No module named 'scapy'"

After some analysis i understand that if you run pip install with a regular user AND run the script with root (required by scapy) the previously installed lib was available only to the user and not to the root. Because of that i needed to rerun the install process also with root.

`sudo pip install scapy `

after that all works!

**Comment by zhy2on** [comment]
> I had this problem because I was naming my file scapy.py and trying to run it using python scapy.py. That didn't work, but changing the name to just scapy, making it executable, and just running it with ./scapy seems to work

Thanks!!!!
